### PR TITLE
dev-util/conan: update and fixes

### PR DESCRIPTION
**Pull request by winterheart**
* Updated to 1.19.2 (#696404)
* Enabled tests (#692036)

### CODE REVIEW

File: dev-util/conan/conan-1.19.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{5,6} )
```
Comment (juippis) [inline dev-util/conan/conan-1.19.2.ebuild]: Any chance you could try with 3.7? Often just adding it, then running repoman, is good indicator if its supported or not. Then you have python_test enabled now, so its easy to test whether it works or not, if deps are satisfied.

File: dev-util/conan/conan-1.19.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	>=dev-python/pylint-2.3.1[${PYTHON_USEDEP}]
+	>=dev-python/python-dateutil-2.7.0[${PYTHON_USEDEP}]
+	>=dev-python/pyyaml-5.1[${PYTHON_USEDEP}]
+	>=dev-python/requests-2.8.1[${PYTHON_USEDEP}]
+	>=dev-python/six-1.10[${PYTHON_USEDEP}]
+	>=dev-python/tqdm-4.28.1[${PYTHON_USEDEP}]
+	>=dev-python/typed-ast-1.1.0[${PYTHON_USEDEP}]
+"
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
```
Comment (juippis) [inline dev-util/conan/conan-1.19.2.ebuild]: With EAPI-7 this is a BDEPEND, and already handled by distutils-r1.eclass. So you can remove it from here.

File: dev-util/conan/conan-1.19.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	test? (
+		${RDEPEND}
+		dev-lang/go
```
Comment (juippis) [inline dev-util/conan/conan-1.19.2.ebuild]: I think this and the cmake below are also a BDEPEND even in test phase, when using EAPI-7. So just add `BDEPEND="test? ( ...`

File: dev-util/conan/conan-1.19.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+"
+
+src_prepare() {
+	default
+	# Fix strict dependencies
+	sed -i \
+		-e "s:patch==1.16:patch>=1.16:g" \
+		-e "s:node-semver==0.6.1:node-semver>=0.6.1:g" \
+		-e "s:future==0.16.0:future>=0.16.0:g" \
+		conans/requirements.txt
```
Comment (juippis) [inline dev-util/conan/conan-1.19.2.ebuild]: Need to append `|| die` here.

File: dev-util/conan/conan-1.19.2.ebuild (hunk 1 above)
Comment (winterheart) [reply dev-util/conan/conan-1.19.2.ebuild]: dev-python/patch don't have python3_7, so no luck

**Comment by winterheart** [comment]
@juippis  done.

**Comment by winterheart** [comment]
@juippis done.

**Pull request closed without merging**